torch/csrc/jit/tensorexpr/expr.h

```diff
@@ -9,11 +9,48 @@ namespace torch {
 namespace jit {
 namespace tensorexpr {
 
+class KernelObject;
+class Kernel {
+ public:
+  static Kernel& GetCurrentKernel();
+  TORCH_API Kernel() {}
+  TORCH_API ~Kernel();
+
+ private:
+  Kernel(const Kernel&) = delete;
+  Kernel& operator=(const Kernel&) = delete;
+  friend class KernelObject;
+  std::vector<KernelObject*> kernel_objects_;
+};
+
+class KernelScope {
+ public:
+  TORCH_API KernelScope();
+  TORCH_API explicit KernelScope(Kernel& kernel);
+  TORCH_API ~KernelScope();
+
+ private:
+  KernelScope(const KernelScope&) = delete;
+  KernelScope& operator=(const KernelScope&) = delete;
+  bool owning_kernel_ = false;
+  Kernel* kernel_ = nullptr;
+};
+
+class TORCH_API KernelObject {
```

Comment:
minor, can we name this something like `KernelScopedObject`?  On initial read I thought this was an object representing a kernel and was confused why it and Kernel both existed.

Reply:
Done in a new PR #94. Thanks!